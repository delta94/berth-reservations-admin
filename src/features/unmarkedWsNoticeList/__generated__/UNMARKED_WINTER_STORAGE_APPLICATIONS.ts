/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ApplicationStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: UNMARKED_WINTER_STORAGE_APPLICATIONS
// ====================================================

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node_customer {
  __typename: "ProfileNode";
  id: string;
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node_winterStorageAreaChoices {
  __typename: "WinterStorageAreaChoiceType";
  priority: number;
  winterStorageAreaName: string;
  winterStorageArea: string | null;
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node {
  __typename: "WinterStorageApplicationNode";
  id: string;
  status: ApplicationStatus;
  createdAt: any;
  municipality: string;
  customer: UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node_customer | null;
  boatType: string | null;
  boatRegistrationNumber: string;
  boatWidth: number;
  boatLength: number;
  boatName: string;
  boatModel: string;
  winterStorageAreaChoices: (UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node_winterStorageAreaChoices | null)[] | null;
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges {
  __typename: "WinterStorageApplicationNodeEdge";
  node: UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node | null;
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications {
  __typename: "WinterStorageApplicationNodeConnection";
  count: number;
  edges: (UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges | null)[];
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS_boatTypes {
  __typename: "BoatTypeType";
  id: string;
  name: string | null;
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONS {
  winterStorageApplications: UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications | null;
  boatTypes: UNMARKED_WINTER_STORAGE_APPLICATIONS_boatTypes[] | null;
}

export interface UNMARKED_WINTER_STORAGE_APPLICATIONSVariables {
  first: number;
  after?: string | null;
  orderBy?: string | null;
}
