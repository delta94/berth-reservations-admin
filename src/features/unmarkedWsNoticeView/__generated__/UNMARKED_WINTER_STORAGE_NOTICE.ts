/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { BerthApplicationLanguage, CustomerGroup, InvoicingType, OrganizationType, Language, ApplicationStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: UNMARKED_WINTER_STORAGE_NOTICE
// ====================================================

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_organization {
  __typename: "OrganizationNode";
  address: string;
  businessId: string;
  city: string;
  name: string;
  organizationType: OrganizationType;
  postalCode: string;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_primaryAddress {
  __typename: "AddressNode";
  address: string;
  postalCode: string;
  city: string;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_primaryEmail {
  __typename: "EmailNode";
  email: string;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_primaryPhone {
  __typename: "PhoneNode";
  phone: string | null;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer {
  __typename: "ProfileNode";
  customerGroup: CustomerGroup | null;
  comment: string | null;
  firstName: string;
  invoicingType: InvoicingType | null;
  lastName: string;
  id: string;
  organization: UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_organization | null;
  primaryAddress: UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_primaryAddress | null;
  primaryEmail: UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_primaryEmail | null;
  primaryPhone: UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer_primaryPhone | null;
  language: Language | null;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_winterStorageAreaChoices {
  __typename: "WinterStorageAreaChoiceType";
  winterStorageArea: string | null;
  priority: number;
  winterStorageAreaName: string;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication {
  __typename: "WinterStorageApplicationNode";
  id: string;
  firstName: string;
  lastName: string;
  address: string;
  municipality: string;
  zipCode: string;
  phoneNumber: string;
  email: string;
  businessId: string;
  companyName: string;
  language: BerthApplicationLanguage;
  customer: UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_customer | null;
  createdAt: any;
  boatType: string | null;
  boatRegistrationNumber: string;
  boatWidth: number;
  boatLength: number;
  boatName: string;
  boatModel: string;
  acceptBoatingNewsletter: boolean;
  acceptFitnessNews: boolean;
  acceptLibraryNews: boolean;
  acceptOtherCultureNews: boolean;
  status: ApplicationStatus;
  winterStorageAreaChoices: (UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication_winterStorageAreaChoices | null)[] | null;
  applicationCode: string;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE_boatTypes {
  __typename: "BoatTypeType";
  id: string;
  name: string | null;
}

export interface UNMARKED_WINTER_STORAGE_NOTICE {
  winterStorageApplication: UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication | null;
  boatTypes: UNMARKED_WINTER_STORAGE_NOTICE_boatTypes[] | null;
}

export interface UNMARKED_WINTER_STORAGE_NOTICEVariables {
  id: string;
}
