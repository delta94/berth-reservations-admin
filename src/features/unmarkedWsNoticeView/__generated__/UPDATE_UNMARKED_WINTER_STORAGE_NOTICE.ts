/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UpdateWinterStorageApplicationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: UPDATE_UNMARKED_WINTER_STORAGE_NOTICE
// ====================================================

export interface UPDATE_UNMARKED_WINTER_STORAGE_NOTICE_updateWinterStorageApplication_winterStorageApplication_customer {
  __typename: "ProfileNode";
  id: string;
}

export interface UPDATE_UNMARKED_WINTER_STORAGE_NOTICE_updateWinterStorageApplication_winterStorageApplication {
  __typename: "WinterStorageApplicationNode";
  id: string;
  customer: UPDATE_UNMARKED_WINTER_STORAGE_NOTICE_updateWinterStorageApplication_winterStorageApplication_customer | null;
}

export interface UPDATE_UNMARKED_WINTER_STORAGE_NOTICE_updateWinterStorageApplication {
  __typename: "UpdateWinterStorageApplicationPayload";
  winterStorageApplication: UPDATE_UNMARKED_WINTER_STORAGE_NOTICE_updateWinterStorageApplication_winterStorageApplication | null;
}

export interface UPDATE_UNMARKED_WINTER_STORAGE_NOTICE {
  updateWinterStorageApplication: UPDATE_UNMARKED_WINTER_STORAGE_NOTICE_updateWinterStorageApplication | null;
}

export interface UPDATE_UNMARKED_WINTER_STORAGE_NOTICEVariables {
  input: UpdateWinterStorageApplicationInput;
}
