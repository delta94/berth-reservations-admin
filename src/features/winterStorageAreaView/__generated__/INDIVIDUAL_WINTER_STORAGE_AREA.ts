/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { LeaseStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: INDIVIDUAL_WINTER_STORAGE_AREA
// ====================================================

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_maps {
  __typename: "WinterStorageAreaMapType";
  id: any;
  url: string;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges_node_application_customer {
  __typename: "ProfileNode";
  id: string;
  firstName: string;
  lastName: string;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges_node_application {
  __typename: "WinterStorageApplicationNode";
  id: string;
  createdAt: any;
  customer: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges_node_application_customer | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges_node {
  __typename: "WinterStorageLeaseNode";
  id: string;
  startDate: any;
  endDate: any;
  status: LeaseStatus;
  isActive: boolean;
  application: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges_node_application | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges {
  __typename: "WinterStorageLeaseNodeEdge";
  node: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges_node | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases {
  __typename: "WinterStorageLeaseNodeConnection";
  edges: (INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases_edges | null)[];
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges_node_application_customer {
  __typename: "ProfileNode";
  id: string;
  firstName: string;
  lastName: string;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges_node_application {
  __typename: "WinterStorageApplicationNode";
  id: string;
  createdAt: any;
  customer: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges_node_application_customer | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges_node {
  __typename: "WinterStorageLeaseNode";
  id: string;
  startDate: any;
  endDate: any;
  status: LeaseStatus;
  isActive: boolean;
  application: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges_node_application | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges {
  __typename: "WinterStorageLeaseNodeEdge";
  node: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges_node | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases {
  __typename: "WinterStorageLeaseNodeConnection";
  edges: (INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases_edges | null)[];
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node {
  __typename: "WinterStoragePlaceNode";
  id: string;
  number: number;
  width: number;
  length: number;
  isActive: boolean;
  leases: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node_leases | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges {
  __typename: "WinterStoragePlaceNodeEdge";
  node: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places {
  __typename: "WinterStoragePlaceNodeConnection";
  edges: (INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges | null)[];
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties {
  __typename: "WinterStorageSectionProperties";
  identifier: string;
  electricity: boolean;
  water: boolean;
  gate: boolean;
  summerStorageForBoats: boolean;
  summerStorageForTrailers: boolean;
  summerStorageForDockingEquipment: boolean;
  leases: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases | null;
  places: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node {
  __typename: "WinterStorageSectionNode";
  id: string;
  properties: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges {
  __typename: "WinterStorageSectionNodeEdge";
  node: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections {
  __typename: "WinterStorageSectionNodeConnection";
  edges: (INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges | null)[];
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties {
  __typename: "WinterStorageAreaProperties";
  name: string | null;
  zipCode: string;
  municipality: string | null;
  streetAddress: string | null;
  wwwUrl: string;
  imageFile: string | null;
  maps: (INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_maps | null)[];
  sections: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea {
  __typename: "WinterStorageAreaNode";
  properties: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREA {
  winterStorageArea: INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea | null;
}

export interface INDIVIDUAL_WINTER_STORAGE_AREAVariables {
  id: string;
}
