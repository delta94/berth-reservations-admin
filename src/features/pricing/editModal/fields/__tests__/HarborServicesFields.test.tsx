import React from 'react';
import { mount } from 'enzyme';
import { Form, Formik } from 'formik';

import HarborServicesFields from '../HarborServicesFields';
import { HarborService } from '../../../harborServicePricing/HarborServicePricing';
import { PeriodType, PriceUnits, ProductServiceType } from '../../../../../@types/__generated__/globalTypes';

const mockValues: HarborService = {
  id: '1',
  service: ProductServiceType.MOORING,
  price: 28,
  unit: PriceUnits.PERCENTAGE,
  period: PeriodType.SEASON,
};

describe('HarborServicesFields', () => {
  const getWrapper = () =>
    mount(
      <Formik initialValues={mockValues} onSubmit={jest.fn()} validate={jest.fn()}>
        {() => (
          <Form>
            <HarborServicesFields />
          </Form>
        )}
      </Formik>
    ).find(HarborServicesFields);

  it('renders normally', () => {
    const wrapper = getWrapper();
    expect(wrapper.render()).toMatchSnapshot();
  });

  describe('Service field', () => {
    it('Should show provided initialValue', () => {
      const input = getWrapper().find('Select').at(0);
      expect(input.prop('id')).toEqual('service');
      expect(input.prop('value')).toEqual('MOORING');
    });
  });

  describe('Price field', () => {
    it('Should show provided initialValue', () => {
      const input = getWrapper().find('input#price');
      expect(input.prop('value')).toEqual(28);
    });
  });

  describe('Unit field', () => {
    it('Should show provided initialValue', () => {
      const input = getWrapper().find('Select').at(1);
      expect(input.prop('id')).toEqual('unit');
      expect(input.prop('value')).toEqual('PERCENTAGE');
    });
  });

  describe('Period field', () => {
    it('Should show provided initialValue', () => {
      const input = getWrapper().find('Select').at(2);
      expect(input.prop('id')).toEqual('period');
      expect(input.prop('value')).toEqual('SEASON');
    });
  });
});
