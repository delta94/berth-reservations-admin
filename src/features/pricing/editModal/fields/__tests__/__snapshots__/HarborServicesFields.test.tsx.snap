// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HarborServicesFields renders normally 1`] = `
Array [
  <div
    class="row"
  >
    <div>
      <div
        class="formTypeTitleLabel"
      >
        Tietue
      </div>
      <div
        class="formTypeTitleValue"
      >
        Satamapalvelut
      </div>
    </div>
  </div>,
  <hr />,
  <div
    class="grid cols1 row"
  >
    <div
      class="Dropdown-module_root__2LI4- Dropdown-module_disabled__2ej9O select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="service-toggle-button"
        id="service-label"
      >
        Palvelu
        <span
          class="RequiredIndicator-module_indicator__Pt7GX"
        >
          *
        </span>
      </label>
      <div
        class="Dropdown-module_wrapper__1oIw5"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="service-label service-toggle-button"
          class="Dropdown-module_buttonDropdown__IwbnQ Dropdown-module_dropdown__22ZdF"
          disabled=""
          id="service-toggle-button"
          type="button"
        >
          <span
            class="Dropdown-module_buttonLabel__2jVRU"
          >
            Kiinnitys
          </span>
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Dropdown-module_angleIcon__1Y1mT"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
      </div>
      <ul
        aria-labelledby="service-label"
        class="Dropdown-module_menu__3rei6 Dropdown-module_overflow__I5ddb"
        id="service-menu"
        role="listbox"
        style="max-height: calc(var(--dropdown-height) * 5);"
        tabindex="-1"
      />
    </div>
  </div>,
  <div
    class="grid cols2 row"
  >
    <div
      class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="price"
      >
        Hinta
        <span
          class="RequiredIndicator-module_indicator__Pt7GX"
        >
          *
        </span>
      </label>
      <div
        class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
      >
        <input
          class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
          id="price"
          name="price"
          required=""
          type="text"
          value="28"
        />
      </div>
    </div>
    <div
      class="Dropdown-module_root__2LI4- select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="unit-toggle-button"
        id="unit-label"
      >
        Yksikk√∂
      </label>
      <div
        class="Dropdown-module_wrapper__1oIw5"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="unit-label unit-toggle-button"
          class="Dropdown-module_buttonDropdown__IwbnQ Dropdown-module_dropdown__22ZdF"
          id="unit-toggle-button"
          type="button"
        >
          <span
            class="Dropdown-module_buttonLabel__2jVRU"
          >
            %
          </span>
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Dropdown-module_angleIcon__1Y1mT"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
      </div>
      <ul
        aria-labelledby="unit-label"
        class="Dropdown-module_menu__3rei6"
        id="unit-menu"
        role="listbox"
        style="max-height: calc(var(--dropdown-height) * 5);"
        tabindex="-1"
      />
    </div>
  </div>,
  <div
    class="grid cols2 row"
  >
    <div
      class="Dropdown-module_root__2LI4- Dropdown-module_disabled__2ej9O select"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="period-toggle-button"
        id="period-label"
      >
        Aika
      </label>
      <div
        class="Dropdown-module_wrapper__1oIw5"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="period-label period-toggle-button"
          class="Dropdown-module_buttonDropdown__IwbnQ Dropdown-module_dropdown__22ZdF"
          disabled=""
          id="period-toggle-button"
          type="button"
        >
          <span
            class="Dropdown-module_buttonLabel__2jVRU"
          >
            Kausi
          </span>
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Dropdown-module_angleIcon__1Y1mT"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
      </div>
      <ul
        aria-labelledby="period-label"
        class="Dropdown-module_menu__3rei6"
        id="period-menu"
        role="listbox"
        style="max-height: calc(var(--dropdown-height) * 5);"
        tabindex="-1"
      />
    </div>
  </div>,
]
`;
